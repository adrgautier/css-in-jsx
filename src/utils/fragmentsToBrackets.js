import React from 'react';

const fragmentsToBrackets = (element) => {
    let outputStyle = element;

    if(!React.isValidElement(element)) {
        return outputStyle;
    }

    outputStyle = '';

    React.Children.forEach(element.props.children, child => {
        const childOutputStyle = fragmentsToBrackets(child);
        outputStyle += childOutputStyle;
    })

    return `{${outputStyle}}`;
};

export default fragmentsToBrackets;