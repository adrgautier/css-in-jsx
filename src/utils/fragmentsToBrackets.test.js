import React from "react";
import fragmentsToBrackets from "./fragmentsToBrackets";

describe("fragmentsToBrackets", () => {
  it("should match snapshot", () => {
    // GIVEN
    const styleAsFragments = (
      <>
        color: black; .active<>color: red;</>
      </>
    );

    // WHEN
    const styleAsBrackets = fragmentsToBrackets(styleAsFragments);

    // THEN
    expect(styleAsBrackets).toMatchSnapshot();
  });
});
