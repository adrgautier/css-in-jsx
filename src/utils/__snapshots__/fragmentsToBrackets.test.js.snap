// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fragmentsToBrackets should match snapshot 1`] = `"{color: black; .active{color: red;}}"`;
